ui/app/common/domain/services/diagnosisService.js

```diff
@@ -1,11 +1,12 @@
 'use strict';
 
 angular.module('bahmni.common.domain')
-    .service('diagnosisService', ['$http', '$rootScope', function ($http, $rootScope) {
+    .service('diagnosisService', ['$http', '$rootScope', 'appService', function ($http, $rootScope, appService) {
         var self = this;
         this.getAllFor = function (searchTerm, locale) {
             var url = Bahmni.Common.Constants.emrapiConceptUrl;
-            var parameters = { term: searchTerm, limit: Bahmni.Common.Constants.emrapiDiagnosisLimit };
+            var searchLimit = appService.getAppDescriptor().getConfigValue("diagnosisSugesstionLimit");
```

Comment:
if this config is not present, then it should get the value from the constant Bahmni.Common.Constants.emrapiDiagnosisLimit